subsys/caf/modules/ble_adv.c

```diff
@@ -598,6 +676,37 @@ static void fast_adv_end_fn(struct k_work *work)
 	__ASSERT_NO_MSG(!fast_adv);
 }
 
+static void rpa_rotate_fn(struct k_work *work)
+{
+	ARG_UNUSED(work);
+
+	if (IS_ENABLED(CONFIG_CAF_BLE_ADV_GRACE_PERIOD) && (state == STATE_GRACE_PERIOD)) {
+		(void)k_work_cancel_delayable(&grace_period_end);
+		grace_period_end_fn(NULL);
+		return;
+	}
+
+	if (state == STATE_DELAYED_ACTIVE) {
+		/* Cannot start advertising now. New advertising session will be started soon. */
+		__ASSERT_NO_MSG(req_new_adv_session);
+		return;
+	}
+
+	__ASSERT_NO_MSG(state == STATE_ACTIVE);
+
+	int err = bt_le_adv_stop();
+
+	if (!err) {
+		__ASSERT_NO_MSG(!req_new_adv_session);
+		force_rpa_rotation(cur_identity);
+		ble_adv_start();
+	}
+
+	if (err) {
+		module_set_state(MODULE_STATE_ERROR);
+	}
```

Comment:
Maybe we can return early in case of an error instead?

Reply:
The intent is to keep this error-handling line at the very end of the function and to add new actions before it in this ```if (!err)``` nRF Desktop fashion.